src/schema.rs

```diff
@@ -352,10 +378,115 @@ fn overwrite_toml_output(
+    if let Some(code) = status.code() {
+        if status.success() {
+            if match doc.get("status") {
+                Some(toml_edit::Item::Value(toml_edit::Value::String(ref expected)))
+                    if expected.value() == "success" =>
+                {
+                    false
+                }
+                Some(
+                    toml_edit::Item::Value(toml_edit::Value::InlineTable(_))
+                    | toml_edit::Item::Table(_),
+                ) => !matches!(
+                    doc["status"].get("code"),
+                    Some(toml_edit::Item::Value(toml_edit::Value::Integer(ref expected)))
+                        if expected.value() == &0),
+                _ => true,
+            } {
+                doc["status"] = toml_edit::Item::None;
+            }
+        } else {
+            let code = code as i64;
+            match doc.get("status") {
+                Some(toml_edit::Item::Value(toml_edit::Value::String(ref expected))) => {
+                    if expected.value() != "failed" {
+                        doc["status"] = toml_edit::value("failed");
+                    }
+                }
+                Some(
+                    toml_edit::Item::Value(toml_edit::Value::InlineTable(_))
+                    | toml_edit::Item::Table(_),
+                ) => {
+                    if !matches!(
+                        doc["status"].get("code"),
+                        Some(toml_edit::Item::Value(toml_edit::Value::Integer(ref expected)))
+                            if expected.value() == &code)
+                    {
+                        doc["status"]["code"] = toml_edit::value(code);
+                    }
+                }
+                _ => {
+                    doc["status"] = toml_edit::value(toml_edit::InlineTable::default());
+                    doc["status"]["code"] = toml_edit::value(code);
+                }
+            }
+        }
+    } else if !matches!(
+        doc.get("status"),
+        Some(toml_edit::Item::Value(toml_edit::Value::String(ref expected)))
+            if expected.value() == "interrupted")
+    {
+        doc["status"] = toml_edit::value("interrupted");
+    }
+
+    Ok(doc.to_string())
+}
+
+fn overwrite_trycmd_status(
+    exit: Option<ExitStatus>,
+    expected_status: Option<CommandStatus>,
+    expected_status_source: Option<usize>,
+    line_nums: &mut Range<usize>,
+    normalized: &mut String,
+) -> Result<(), snapbox::Error> {
+    if let Some(status) = exit {
+        if let Some(status) = if let Some(code) = status.code() {
+            if status.success() {
+                if let (true, Some(line_num)) = (
+                    expected_status != Some(CommandStatus::Success),
+                    expected_status_source,
+                ) {
+                    replace_lines(normalized, line_num..line_num + 1, "")?;
+                    *line_nums = line_nums.start - 1..line_nums.end - 1;
```

Comment:
Why do we need to change these?

Reply:
since we deleted one line, the stdout overwrites need to be shifted upwards by one line

Reply:
If `line_nums` is specifically the lines for stdout, maybe we should put that in the name?

I also feel like either a function or at least parenthesis are needed as the formatting makes it look like the range is `1..line_nums.end`

Reply:
The default formatting with no spaces is a bit confusing, I have added the parenthesis for ranges